components/Banner/src/MDCBannerView.m

```diff
@@ -0,0 +1,474 @@
+      frameHeight += kTopPaddingLarge + kBottomPadding;
+      frameHeight += [self getFrameHeightOfImageViewAndTextLabelWithSizeToFit:size];
+      CGSize leadingButtonSize = [self.leadingButton sizeThatFits:CGSizeZero];
+      CGSize trailingButtonSize = [self.trailingButton sizeThatFits:CGSizeZero];
+      frameHeight += MAX(leadingButtonSize.height, trailingButtonSize.height);
+      break;
+    }
+    case MDCBannerViewLayoutModeMultiLineStackedButton: {
+      frameHeight += kTopPaddingLarge + kBottomPadding;
+      frameHeight += [self getFrameHeightOfImageViewAndTextLabelWithSizeToFit:size];
+      CGSize leadingButtonSize = [self.leadingButton sizeThatFits:CGSizeZero];
+      CGSize trailingButtonSize = [self.trailingButton sizeThatFits:CGSizeZero];
+      frameHeight +=
+          leadingButtonSize.height + trailingButtonSize.height + kButtonVerticalIntervalSpace;
+      break;
+    }
+  }
+  return CGSizeMake(size.width, frameHeight);
+}
+
+- (void)updateConstraints {
+  if (@available(iOS 9.0, *)) {
+    MDCBannerViewLayoutMode layoutMode = [self layoutModeForSizeToFit:self.bounds.size];
+    [self updateConstraintsWithLayoutMode:layoutMode];
+  }
+
+  [super updateConstraints];
+}
+
+#pragma mark - Layout methods
+
+- (void)updateConstraintsWithLayoutMode:(MDCBannerViewLayoutMode)layoutMode NS_AVAILABLE_IOS(9_0) {
+  [self deactivateAllConstraints];
+  // Set Elements
+  if (layoutMode == MDCBannerViewLayoutModeSingleLine) {
+    if (self.hasImage) {
+      self.imageViewConstraintLeading.active = YES;
+      self.imageViewConstraintTopSmall.active = YES;
+      self.imageViewConstraintBottom.active = YES;
+      self.textLabelConstraintLeadingWithImage.active = YES;
+    } else {
+      self.textLabelConstraintLeadingWithMargin.active = YES;
+    }
+    self.textLabelConstraintCenterY.active = YES;
+    self.buttonContainerConstraintLeadingWithTextLabel.active = YES;
+    self.buttonContainerConstraintLeadingWithTextLabelGreater.active = YES;
+    self.buttonContainerConstraintTrailing.active = YES;
+    self.buttonContainerConstraintTopWithMargin.active = YES;
+    self.buttonContainerConstraintBottom.active = YES;
+  } else {
+    if (self.hasImage) {
+      self.imageViewConstraintLeading.active = YES;
+      self.imageViewConstraintTopLarge.active = YES;
+      self.textLabelConstraintLeadingWithImage.active = YES;
+      self.buttonContainerConstraintTopWithImageViewGreater.active = YES;
+    } else {
+      self.textLabelConstraintLeadingWithMargin.active = YES;
+    }
+    self.textLabelConstraintTop.active = YES;
+    self.textLabelConstraintTrailing.active = YES;
+    self.buttonContainerConstraintTopWithTextLabelGreater.active = YES;
+    self.buttonContainerConstraintTopWithTextLabel.active = YES;
+    self.buttonContainerConstraintLeading.active = YES;
+    self.buttonContainerConstraintTrailing.active = YES;
+    self.buttonContainerConstraintBottom.active = YES;
```

Comment:
`self.buttonContainerConstraintBottom.active = YES;` is in both of these logical paths. Please extract it out to the top level of this method to clearly indicate that it is common. Similarly for any other repeated logical paths.

Reply:
Addressed in 68bf18bd3a26fbf092e5113a063cb4b4903aa25e.